nixos/modules/services/databases/influxdb2.nix

```diff
@@ -53,6 +110,57 @@ in
         KillMode = "control-group";
         Restart = "on-failure";
       };
+
+      # Mark if this is the first startup so postStart can do the initial setup
+      preStart = mkIf cfg.provision.enable ''
+        if ! test -e "$STATE_DIRECTORY/influxd.bolt"; then
+          touch "$STATE_DIRECTORY/.first_startup"
+        fi
+      '';
+
+      postStart = let
+        initCfg = cfg.provision.initialSetup;
+      in mkIf cfg.provision.enable (
+        ''
+          set -euo pipefail
+          export INFLUX_HOST="http://"${escapeShellArg (cfg.settings.http-bind-address or "localhost:8086")}
+
+          # Wait for the influxdb server to come online
+          count=0
+          while ! ${getExe pkgs.influxdb2-cli} ping &>/dev/null; do
```

Comment:
We can add `influxdb2-cli` to the `PATH` of the service by setting `systemd.services.<name>.path`